django/db/backends/postgresql/operations.py

```diff
@@ -257,3 +258,18 @@ def window_frame_range_start_end(self, start=None, end=None):
                 'and FOLLOWING.'
             )
         return start_, end_
+
+    def explain_query_prefix(self, output_format=None, verbose=False):
+        prefix = super().explain_query_prefix(output_format, verbose)
+
+        extra = {}
+        if output_format:
+            extra['FORMAT'] = output_format
+        if verbose:
+            extra['VERBOSE'] = 'true'
+            extra['ANALYZE'] = 'true'
```

Comment:
I don't think that `ANALYZE` should be set when using `verbose`. This makes it execute the query which could take a long time, especially when users will think this means `VERBOSE`. Perhaps a separate parameter should be used for triggering this behaviour?

Reply:
In addition, use of `ANALYZE` can lead to modification of data which should be documented - even with a `SELECT` query if a function is executed - see the [documentation](https://www.postgresql.org/docs/10/static/sql-explain.html#idm46046850341168). It might make sense to highlight that a transaction should be used which can be aborted if the user wants to avoid this.

Reply:
I didn't consider this, it absolutely should be documented and not the default when `verbose=true`!

Reply:
Something like this comes to mind:

```python
with transaction.atomic():
    transaction.set_rollback(True)
    print(queryset.explain(analyze=True))
```